src/lib/InstantSearch.ts

```diff
@@ -159,14 +156,6 @@ class InstantSearch extends EventEmitter {
       throw new Error(withUsage('The `searchClient` option is required.'));
     }
 
-    if (typeof (options as any).urlSync !== 'undefined') {
-      throw new Error(
-        withUsage(
-          'The `urlSync` option was removed in InstantSearch.js 3. You may want to use the `routing` option.'
```

Comment:
we still have a reasonable amount of people who go from InstantSearch v1 -> v4, can we keep this, but as a development warning?

Reply:
We already removed this error in #4119. I was surprised it was still here. We might have messed up the history.